File: rb/features.rb
# IpGeolocationApi4 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IpGeolocationApi4Features
  def self.make_feature(name)
    case name
    when "base"
      IpGeolocationApi4BaseFeature.new
    when "test"
      IpGeolocationApi4TestFeature.new
    else
      IpGeolocationApi4BaseFeature.new
    end
  end
end

File: py/features.py
# IpGeolocationApi4 SDK feature factory

from feature.base_feature import IpGeolocationApi4BaseFeature
from feature.test_feature import IpGeolocationApi4TestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpGeolocationApi4BaseFeature(),
        "test": lambda: IpGeolocationApi4TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
